src/charts/heatmap.js

```diff
@@ -0,0 +1,350 @@
+                    .append('svg')
+                    .classed('britechart heatmap', true);
+
+                buildContainerGroups();
+            }
+
+            svg
+                .attr('width', width)
+                .attr('height', height);
+        }
+
+        /**
+         * Builds containers for the chart, the axis and a wrapper for all of them
+         * Also applies the Margin convention
+         * @return {void}
+         * @private
+         */
+        function buildContainerGroups() {
+            let container = svg
+              .append('g')
+                .classed('container-group', true)
+                .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+            container
+                .append('g').classed('chart-group', true);
+            container
+                .append('g').classed('day-labels-group', true);
+            container
+                .append('g').classed('hour-labels-group', true);
+            container
+                .append('g').classed('metadata-group', true);
+        }
+
+        /**
+         * Cleaning data casting the values to the proper
+         * type while keeping the rest of properties on the data. It
+         * also creates a set of zeroed data (for animation purposes)
+         * @param   {HeatmapData} originalData  Raw data as passed to the container
+         * @return  {HeatmapData}               Clean data
+         * @private
+         */
+        function cleanData(originalData) {
+            return originalData.reduce((acc, {day, hour, value}) => {
+                return [
+                    ...acc,
+                    {
+                        day: +day,
+                        hour: +hour,
+                        value: +value
+                    }
+                ];
+            }, []);
+        }
+
+        /**
+         * Creates the scales for the heatmap chart
+         * @return void
+         */
+        function buildScales() {
+            colorScale = d3Scale.scaleLinear()
+                .range([colorSchema[0], colorSchema[colorSchema.length - 1]])
+                .domain(d3Array.extent(data, getValue))
+                .interpolate(d3Interpolate.interpolateHcl);
+        }
+
+        /**
+         * Draws the boxes of the heatmap
+         */
+        function drawBoxes() {
+            boxes = svg.select('.chart-group').selectAll('.box').data(data);
+
+            boxes.enter()
+              .append('rect')
+                .classed('box', true)
+                .attr('width', boxSize)
+                .attr('height', boxSize)
+                .attr('x', ({hour}) => hour * boxSize)
+                .attr('y', ({day}) => day * boxSize)
+                .style('opacity', boxInitialOpacity)
+                .style('fill', boxInitialColor)
+                .style('stroke', boxBorderColor)
+                .style('stroke-width', boxBorderSize)
+                .transition()
+                    .duration(animationDuration)
```

Comment:
Is spacing here correct? Doesn't need two spaces instead of a tab?

Reply:
Well, it seems we are not being consistent with this. I have seen no spaces, two spaces and a tab.